### ISSUE TITLE
fix: Accessibility issues in FluentNavMenu

### ISSUE THREAD

**Issue by tlmii** [microsoft/fluentui-blazor#847]
# 🐛 Bug Report

Accessibility Insights shows several issues with the FluentNavMenu. 

1. The `div.fluent-nav-menu` should probably have a `role="menu"` attribute
2. The `div.fluent-nav-item` elements should have a `role="menuitem"` attribute
3. The `div.expand-collapse-button` elements should have `tabindex="-1"` instead of `0` because it is set to `aria-hidden="true"`, or `aria-hidden="true"` should be removed

There might be something that needs to be done with `fluent-nav-group` too, since it has `fluent-nav-item` children, but I'm not sure the right role for it (that error doesn't come up until you start fixing some of the others)

## 💻 Repro or Code Sample
These can be seen on https://www.fluentui-blazor.net/:
![image](https://github.com/microsoft/fluentui-blazor/assets/9613109/aeed085c-061f-4638-a6b4-b3cd39641021)
